builtin/logical/transit/path_cache_config.go

```diff
@@ -70,11 +70,12 @@ func (b *backend) pathCacheConfigWrite(ctx context.Context, req *logical.Request
 		b.lm.ConvertCacheToSyncmap()
 	}
 
+	var err error
 	if cacheType == keysutil.LRU {
-		b.lm.ConvertCacheToLRU(cacheSize)
+		err = b.lm.ConvertCacheToLRU(cacheSize)
```

Comment:
Shouldn't we return a logical.ErrorResponse for the first return value in this scenario (rather than nil)?

Reply:
I am unsure. I was influenced by patterns like [this](https://github.com/hashicorp/vault/blob/master/builtin/logical/transit/path_decrypt.go#L62). Returning `logical.ErrorResponse, nil` will give some feedback to the user, returning `nil, err` will bubble up the error. My gut feeling is to go with what you are suggesting because I think this error should be returned to the user but maybe someone else can provide some input @briankassouf @calvn?

Reply:
Seems that returning a `nil` _*logical.Response_ and a `internal-error` _error_ is the established pattern to default to an _InternalServerError_.

Looking at response_util [here](https://github.com/hashicorp/vault/blob/e9b8dffd763cbc847fd9552992ae8284444483a7/logical/response_util.go#L16) and [here](https://github.com/hashicorp/vault/blob/e9b8dffd763cbc847fd9552992ae8284444483a7/logical/response_util.go#L90).